### PR TITLE
Avoid duplication of image labels, simplify CI code

### PR DESCRIPTION
**Pull request by gtardif**
Replaces #75 , pushed to upstream for CI secrets
Signed-off-by: Guillaume Tardif <guillaume.tardif@gmail.com>

### CODE REVIEW

File: .github/workflows/build-scan-push.yaml [hunk 1]
```diff
@@ -81,30 +81,14 @@ jobs:
...
-          # We have to override them with the same values as the ones defined in the Dockerfile.
-          # https://github.com/docker/metadata-action#overwrite-labels
-          labels: |
-            org.opencontainers.image.title=Volumes Backup & Share
-            org.opencontainers.image.description=Back up, clone, restore, and share Docker volumes effortlessly.
-            org.opencontainers.image.vendor=Docker Inc.
-            org.opencontainers.image.revision=${{ github.event.pull_request.head.sha || github.event.after || github.event.release.tag_name }}
 
       - name: Docker Build and Push to Docker Hub
-        if: ${{ !github.event.pull_request.head.repo.fork }}
```
Comment (felipecruz91) [inline .github/workflows/build-scan-push.yaml:94]: @gtardif I see this condition has been removed. I've changed the repo settings to require approval for all outside contributors to run workflows in our repo.

File: Makefile [hunk 2]
```diff
@@ -13,6 +14,8 @@ build-extension: ## Build service image to be deployed as a desktop extension
 		--secret id=BUGSNAG_API_KEY \
 		--build-arg BUGSNAG_RELEASE_STAGE=$(BUGSNAG_RELEASE_STAGE) \
 		--build-arg BUGSNAG_APP_VERSION=$(TAG) \
+		--label org.opencontainers.image.source=https://github.com/docker/volume-backup-extension \
+		--label org.opencontainers.image.revision=$(GIT_COMMIT) \
```
Comment (felipecruz91) [inline Makefile:18]: Apparently, the `GIT_COMMIT` doesn't work in PRs as it leaves the CI in a waiting state forever: 

`docker-vulnerability-policy Expected — Waiting for status to be reported`.

We need to keep `${{ github.event.pull_request.head.sha || github.event.after || github.event.release.tag_name }}` as the value for the `org.opencontainers.image.revision` label. 

I'll add a commit with the fix.

File: .github/workflows/build-scan-push.yaml [hunk 3]
```diff
@@ -81,30 +81,14 @@ jobs:
...
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
-          platforms: linux/amd64,linux/arm64
-          build-args: |
-            BUGSNAG_RELEASE_STAGE=production
-            BUGSNAG_APP_VERSION=${{ github.event.release.tag_name }}
-          secrets: |
-            BUGSNAG_API_KEY=${{ secrets.BUGSNAG_API_KEY }}
+        env:
+          TAG: ${{ steps.docker_meta.outputs.version }}
```
Comment (felipecruz91) [inline .github/workflows/build-scan-push.yaml:87]:
```suggestion
          GIT_COMMIT: ${{ github.event.pull_request.head.sha || github.event.after || github.event.release.tag_name }}
          TAG: ${{ steps.docker_meta.outputs.version }}
```

**Comment by gtardif** [comment]
7 mins to rebuild everything, this is because we loose the buildx cache with this PR.
I'll find some other way to reduce label duplication, but this is too bad, 7 mins for every docker build!

**Pull request closed without merging**